README.md

```diff
@@ -831,6 +831,338 @@ to the `assets/css/app.css` file:
+     text-white bg-gradient-to-r to-[#b6c5ff] from-[#e07db3]
+     text-base font-semibold py-2 px-4 rounded-lg shadow-xl
+     hover:bg-pink-700 focus:outline-none focus:ring-2 focus:ring-blue-500 focus:ring-offset-2 focus:ring-offset-pink-200"
+    value="Login"
+  />
+
+  <ul class="list-none mt-9 max-h-full max-w-full" id="cursor" phx-hook="TrackClientCursor">
+    <%= for user <- @users do %>
+      <li style={"left: #{user.x}%; top: #{user.y}%"} class="flex flex-col absolute pointer-events-none whitespace-nowrap overflow-hidden">
+        <svg xmlns="http://www.w3.org/2000/svg" width="31" height="32" fill="none" viewBox="0 0 31 32">
+          <path fill="url(#a)" d="m.609 10.86 5.234 15.488c1.793 5.306 8.344 7.175 12.666 3.612l9.497-7.826c4.424-3.646 3.69-10.625-1.396-13.27L11.88 1.2C5.488-2.124-1.697 4.033.609 10.859Z"/>
+          <defs>
+            <linearGradient id="a" x1="-4.982" x2="23.447" y1="-8.607" y2="25.891" gradientUnits="userSpaceOnUse">
+              <stop style={"stop-color: #{user.color}"}/>
+              <stop offset="1" stop-color="#BDACFF"/>
+            </linearGradient>
+          </defs>
+        </svg>
+        <span style={"background-color: #{user.color};"} class="mt-1 ml-4 px-1 text-sm text-white rounded-xl">
+          <%= user.username %>
+        </span>
+      </li>
+    <% end %>
+  </ul>
+</div>
+```
+
+We've simply added a button that,
+once clicked, creates a `login` event.
+We are going to eventually handle this event,
+but before that,
+we need to add `auth_plug` to our project.
+
+### 5b. Add `auth_plug`
+
+Let's get this up and running fast!
+Inside the `mix.exs` file,
+add the following inside the `deps` section.
+
+```elixir
+{:auth_plug, "~> 1.5"},
+```
+
+and run 
+
+```sh
+mix deps.get
+```
+
+Afterwards, visit https://authdemo.fly.dev/apps/new,
+sign in using a preferred provider 
+and create an app for the `localhost:4000` URL,
+the same URL you are running the app from.
+
+![flyapp](https://user-images.githubusercontent.com/17494745/203396767-a73a6c5e-f484-49a2-9ece-225e5b09f740.png)
+
+After creating an app, 
+you should be prompted with an image similar
+to the following.
+
+![token](https://user-images.githubusercontent.com/194400/202419365-e06c43d6-c537-4646-a1a3-90320fb3aa59.png)
+
+You now are going to run the app with this `AUTH_API_KEY`
+environment variable. 
+To run the app with this environment variable,
+
+```sh
+export AUTH_API_KEY=2cfxNaMmkvwKmHncbYAL58mLZMs/2cfxNa4RnU12gYYSwPvp2hSPFdVDcbdK/authdemo.fly.dev
+
+mix phx.server
+```
+
+Now, let's create the `AuthController`.
+This controller will be used to login 
+and logout the user in the app.
+
+Create a file `lib/chat_web/controllers/auth_controller.ex`
+and add the following.
+
+```elixir
+defmodule LiveCursorsWeb.AuthController do
+  use LiveCursorsWeb, :controller
+  import Phoenix.LiveView, only: [assign_new: 3]
+
+  def add_assigns(:default, _params, %{"jwt" => jwt} = _session, socket) do
```

Comment:
This code should be DRYed: https://github.com/dwyl/auth_plug/issues/86 
Copy-pasting it from the MVP is a clear sign that it should be included in the library. 💭

Reply:
Created this PR -> https://github.com/dwyl/auth_plug/pull/88. 
After merging and publishing a new `auth-plug` version, I can make the changes and deploy the "Live Cursors" with auth version to fly.io.

Reply:
As of `auth_plug@1.5.1` you can now call: 

```elixir
socket = socket
  |> AuthPlug.assign_jwt_to_socket(&Phoenix.LiveView.assign_new/3, jwt)
```

Reply:
Or ... in-line the whole call as:

```elixir
def add_assigns(:default, _params, %{"jwt" => jwt} = _session, socket) do
  {:cont,  
    AuthPlug.assign_jwt_to_socket(socket, &Phoenix.LiveView.assign_new/3, jwt)}
end
```

Reply:
The instance in the `code` is fixed by https://github.com/dwyl/phoenix-liveview-realtime-cursor-tracking-tutorial/commit/a75ef36eeb16c0b3918b01f3a2a9d4635ba9475b but not the instructions (`code` block) in the `README.md` ... 💭

Reply:
I'm dumb.
Fixed with https://github.com/dwyl/phoenix-liveview-realtime-cursor-tracking-tutorial/pull/12/commits/84f1687d2e74b43072c3fff7485293c17da83382

sorry

Reply:
Not "dumb" at all. We just have the code in two places in these tutorials ... 🙄 
If there was a way to DRY this e.g. using `LiveBook` for all our tutorials/examples: https://github.com/dwyl/learn-phoenix-framework/issues/147 📖 
Then we could avoid the duplication. 💭